core/src/ast/types.rs

```diff
@@ -105,14 +105,61 @@ pub enum ModSymbol {
     Trait(Trait),
 }
 
-/// A named type or trait that is just a path, e.g. `std::borrow::Cow<'a, T>`.
+/// A named type that is just a path, e.g. `std::borrow::Cow<'a, T>`.
 #[derive(Clone, PartialEq, Eq, Hash, Serialize, Deserialize, Debug)]
 #[non_exhaustive]
 pub struct PathType {
     pub path: Path,
     pub lifetimes: Vec<Lifetime>,
 }
 
+/// A named trait that is just a path, e.g. `std::borrow::Cow<'a, T>`.
+#[derive(Clone, PartialEq, Eq, Hash, Serialize, Deserialize, Debug)]
+#[non_exhaustive]
+pub struct PathTrait {
```

Comment:
thought: I am okay with PathType and PathTrait being the same, but this is ultimately fine too

Reply:
i think it's nice having them separate b/c the `from_syn` for types takes a `syn::TypePath`, and for traits it has to be something else (i made it `syn::TraitBound`)

Reply:
Oh, good point. Youre right!

Reply:
Hmm actually, on second thought, I still think it makes sense to be separate -- there needs to be `to_syn` that resolves to either a `syn` trait or type representation, and the `resolve_with_path` method is also different if the path is a trait or a type.

Reply:
Right, but you can still have two methods with slightly different names for that.